src/http.js

```diff
@@ -1,13 +1,37 @@
 //------------------------------------------------------------------------------
```

Comment:
interesting - is this more like the python http api that you were talking about than the one we require? i like it.